athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/TimeZoneKeyTest.java

```diff
@@ -0,0 +1,248 @@
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.amazonaws.athena.connector.lambda.data;
+
+import com.google.common.collect.ImmutableSortedSet;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+import org.junit.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Comparator;
+import java.util.SortedSet;
+
+import static com.amazonaws.athena.connector.lambda.data.TimeZoneKey.MAX_TIME_ZONE_KEY;
+import static com.amazonaws.athena.connector.lambda.data.TimeZoneKey.UTC_KEY;
+import static java.util.Locale.ENGLISH;
+import static org.junit.Assert.*;
+
+/**
+ * Class for testing timezone to be used in encoding datetime value and timezone value to a single long
+ * a copy from Presto TimeZoneKeyTest
```

Comment:
please ensure we are putting proper attribution to presto for any code that we reuse.